loopchain/blockchain/blockchain.py

```diff
@@ -71,19 +74,19 @@ def __init__(self, channel_name=None, store_id=None):
         self.last_unconfirmed_block = None
         self.__channel_name = channel_name
         self.__peer_id = ChannelProperty().peer_id
+        self.__block_manager: BlockManager = block_manager
 
         store_id = f"{store_id}_{channel_name}"
-        # block db has [ block_hash - block | block_height - block_hash | BlockChain.LAST_BLOCK_KEY - block_hash ]
         self._blockchain_store, self._blockchain_store_path = utils.init_default_key_value_store(store_id)
 
-        # made block count as a leader
-        self.__invoke_results = {}
+        # tx receipts and next prep after invoke, {block_hash: (receipts, next_prep)}
+        self.__invoke_results: AgingCache = AgingCache(max_age_seconds=conf.INVOKE_RESULT_AGING_SECONDS)
```

Comment:
1. How about using Hash32 for BlockHash not Hash32.hex()?

2. Handling it with time seems uncertain.
I recommend
```python
self._tx_results_cache: DefaultDict[int, Dict[Hash32, (dict, Sequence[ExternalAddress])]] = defaultdict(dict)  
receipts, next_preps = self._receipt_cache[block_height][block_hash]  # use case

def trim_cache(self, block_height: int):
    remove_heights = [height_cache for height_cache in self._tx_results_cache
                                     if height_cache <= block_height]
    for remove_height in remove_heights:
        self._tx_results_cache.pop(remove_height, pop)

Reply:
Hash32 is a good suggestion. And this is not a cache. In normal operation, invoke_result always should be empty when processed. But I will consider your recommend later.

Reply:
Assume that a situation invoking many times with diffent block hash in one block height.

Then only 'invoke result' of added block will be removed. Other invoke results remains until the node shuts down.

Reply:
But as you mentioned this PR is not considered of refactoring. Then just keep it . We will be able to improve it later.